From 0e1c336fb73a41ad0a4f1dc6e441b5431d402354 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Fri, 27 Feb 2026 11:22:55 +0000
Subject: [PATCH] Allow system_app to access 8849 projector and laser sysfs

Add SELinux policy to allow TrebleApp (system_app) to read and write
the FPGA SPI sysfs nodes and read the NDS03 laser rangefinder sysfs.
---
 sepolicy/genfs_contexts |  2 ++
 sepolicy/treble_app.te  | 10 ++++++++++
 2 files changed, 12 insertions(+)

diff --git a/sepolicy/genfs_contexts b/sepolicy/genfs_contexts
index 56e8950..17ce43a 100644
--- a/sepolicy/genfs_contexts
+++ b/sepolicy/genfs_contexts
@@ -1 +1,3 @@
 genfscon sysfs /board_properties u:object_r:sysfs_board_properties:s0
+genfscon sysfs /bus/spi/drivers/fpga_spi u:object_r:sysfs_fpga_spi:s0
+genfscon sysfs /nds03_ctrl u:object_r:sysfs_nds03:s0
diff --git a/sepolicy/treble_app.te b/sepolicy/treble_app.te
index 4bed5f1..dc7288f 100644
--- a/sepolicy/treble_app.te
+++ b/sepolicy/treble_app.te
@@ -39,3 +39,13 @@ binder_call(system_app, hal_gesturewake_default);
 
 type hal_aidl_agolddaemon_default, domain;
 binder_call(system_app, hal_aidl_agolddaemon_default);
+
+# Unihertz 8849 (Tank) projector FPGA sysfs access.
+type sysfs_fpga_spi, fs_type, sysfs_type;
+allow system_app sysfs_fpga_spi:dir r_dir_perms;
+allow system_app sysfs_fpga_spi:file rw_file_perms;
+
+# Unihertz 8849 (Tank) NDS03 laser rangefinder sysfs access.
+type sysfs_nds03, fs_type, sysfs_type;
+allow system_app sysfs_nds03:dir r_dir_perms;
+allow system_app sysfs_nds03:file r_file_perms;
-- 
2.53.0

